btrdb/stream.py

```diff
@@ -2189,23 +2190,37 @@ def arrow_values(
             period_ns = 0
             if sampling_freq > 0:
                 period_ns = _to_period_ns(sampling_freq)
-            params["uu_list"] = [s.uuid for s in self._streams]
-            params["version_list"] = [versions[s.uuid] for s in self._streams]
+
             params["snap_periodNS"] = period_ns
-            table = list(self._btrdb.ep.arrowMultiValues(**params))
-            if len(table) > 0:
-                data = pa.concat_tables(table)
-            else:
-                schema = pa.schema(
-                    [pa.field("time", pa.timestamp("ns", tz="UTC"), nullable=False)]
-                    + [
-                        pa.field(str(s.uuid), pa.float64(), nullable=False)
-                        for s in self._streams
-                    ],
-                )
-                data = pa.Table.from_arrays(
-                    [pa.array([]) for i in range(1 + len(self._streams))], schema=schema
-                )
+            data = None
+            # Batching the multistream queries is a stop gap to prevent
+            # the user from killing the DB by querying the raw values of
+            # too many streams at once.
+            for stream_batch in batched(self._streams, 20000):
```

Comment:
can we make a constant for this limit, `MAX_BATCH_STREAMS = 20000` or something similar to `INSERT_BATCH_SIZE`
```suggestion
            for stream_batch in batched(self._streams, MAX_BATCH_STREAMS):
```

Reply:
Sounds good! Added.